cluster-autoscaler/core/scale_up.go

```diff
@@ -341,7 +341,8 @@ func ScaleUp(context *context.AutoscalingContext, processors *ca_processors.Auto
 
 	nodesFromNotAutoscaledGroups, err := utils.FilterOutNodesFromNotAutoscaledGroups(nodes, context.CloudProvider)
 	if err != nil {
-		return &status.ScaleUpStatus{Result: status.ScaleUpError}, err.AddPrefix("failed to filter out nodes which are from not autoscaled groups: ")
+		err = err.AddPrefix("failed to filter out nodes which are from not autoscaled groups: ")
+		return &status.ScaleUpStatus{Result: status.ScaleUpError, ScaleUpError: &err}, err
```

Comment:
nit: this feels a bit boilerplate-y. Maybe a helper function would be nice? I'm thinking something like `return scaleUpError(err.AddPrefix())`.

I don't feel strongly about this at all, please feel free to ignore this comment if you don't like the idea.

Reply:
Agree with your suggestion, I created scaleUpError() func which takes ScaleUpStatus (it is required as sometimes we pass additional information to ScaleUpStatus) and error as arguments